### DISCUSSION TITLE
Adding a margin to the text in a redaction annotation

### DISCUSSION THREAD

**Discussion by Pmmoks** [pymupdf/PyMuPDF#4148] [Ideas]
Hi,

Is there any way to add a margin (inner) to the text when using `page.add_redact_annot()`? I often find there are slight errors in the conversion of the PDF to markdown if redaction annotations are too close to the edge of the page. But I can't see a direct way of adding a margin for the text within the annotation bbox.

Thanks,

P

**Comment by JorjMcKie** [comment]
No, there is not.
Redactions' sense in life is to remove everything that laps into its rectangle. This even has legal meaning.
After all, this feature has not been introduced to facilitate PDF editing, but to support safe removal of sensitive data.

This may indeed mean that characters are included in the removal for which no **_visual_** overlap exists - but there still may exist some white white space around it.

To avoid this, reduce the redaction rectangle by e.g. using `pymupdf.TOOLS.set_small_glyph_heights(True)` when searching for text, or similar.
